Python/Product/Common/Infrastructure/EnumerableExtensions.cs

```diff
@@ -63,5 +63,25 @@ public static class EnumerableExtensions {
                 yield return new KeyValuePair<TKey, TValue>((TKey)entry.Key, (TValue)entry.Value);
             }
         }
+
+        private class TakeWhileCounter<T> {
+            private ulong _remaining;
+
+            public TakeWhileCounter(ulong count) {
+                _remaining = count;
+            }
+
+            public bool ShouldTake(T value) {
+                if (_remaining == 0) {
+                    return false;
+                }
+                _remaining -= 1;
+                return true;
+            }
+        }
+
+        public static IEnumerable<T> Take<T>(this IEnumerable<T> source, ulong count) {
```

Comment:
Ah, I probably removed them. I think it's okay - these are internal functions and we are likely to legitimately need it somewhere.